react/components/EditorContainer/Sidebar/Content.tsx

```diff
@@ -20,126 +19,102 @@ interface Props {
   iframeRuntime: RenderContext
 }
 
-interface State {
-  components: SidebarComponent[]
-}
-
-class Content extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props)
-
-    this.state = {
-      components: getComponents(
-        props.iframeRuntime.extensions,
-        getIframeRenderComponents(),
-        props.iframeRuntime.page
-      ),
-    }
-  }
-
-  public componentDidUpdate(prevProps: Props) {
-    const prevRuntime = prevProps.iframeRuntime
-    const currRuntime = this.props.iframeRuntime
-
-    const prevPath = prevRuntime.route.path
-    const currPath = currRuntime.route.path
-
-    if (prevPath !== currPath) {
-      this.resetComponents()
-      this.props.editor.setIsLoading(false)
-    }
-  }
+const initialComponents = (props: Props) =>
+  getComponents(
+    props.iframeRuntime.extensions,
+    getIframeRenderComponents(),
+    props.iframeRuntime.page
+  )
 
-  public render() {
-    const { editor, highlightHandler, iframeRuntime } = this.props
+const Content = (props: Props) => {
+  const { editor, highlightHandler, iframeRuntime } = props
 
-    if (editor.editTreePath === null) {
-      return (
-        <ComponentSelector
-          components={this.state.components}
-          editor={editor}
-          highlightHandler={highlightHandler}
-          iframeRuntime={iframeRuntime}
-          updateSidebarComponents={this.updateComponents}
-        />
-      )
-    }
+  const formMeta = useFormMetaContext()
+  const modal = useModalContext()
 
-    const isSitewide = getIsSitewide(
-      iframeRuntime.extensions,
-      editor.editTreePath
+  const [components, setComponents] = useState(
+    getComponents(
+      props.iframeRuntime.extensions,
+      getIframeRenderComponents(),
+      props.iframeRuntime.page
```

Comment:
Perhaps it'd be better to use `initialComponents(props)` (or `getInitialComponents(props)`).